src/ansys/fluent/core/launcher/standalone_launcher.py

```diff
@@ -184,103 +184,111 @@ def __init__(
         del argvals["self"]
         if argvals["start_timeout"] is None:
             argvals["start_timeout"] = 60
-        for arg_name, arg_values in argvals.items():
-            setattr(self, arg_name, arg_values)
-        self.argvals = argvals
-        self.new_session = self.mode.value[0]
+        self.new_session = argvals["mode"].value[0]
         self.file_transfer_service = file_transfer_service
 
-    def __call__(self):
-        if self.lightweight_mode is None:
+        if argvals["lightweight_mode"] is None:
             # note argvals is no longer locals() here due to _get_session_info() pass
-            self.argvals.pop("lightweight_mode")
-            setattr(self, "lightweight_mode", False)
-        fluent_version = _get_standalone_launch_fluent_version(self.product_version)
+            argvals.pop("lightweight_mode")
+            argvals["lightweight_mode"] = False
+        fluent_version = _get_standalone_launch_fluent_version(
+            argvals["product_version"]
+        )
         if fluent_version:
-            _raise_non_gui_exception_in_windows(self.ui_mode, fluent_version)
+            _raise_non_gui_exception_in_windows(argvals["ui_mode"], fluent_version)
 
         if os.getenv("PYFLUENT_FLUENT_DEBUG") == "1":
-            self.argvals["fluent_debug"] = True
+            argvals["fluent_debug"] = True
 
-        server_info_file_name = _get_server_info_file_name()
-        launch_string = _generate_launch_string(
-            self.argvals,
-            self.mode,
-            server_info_file_name,
+        self._server_info_file_name = _get_server_info_file_name()
+        self._launch_string = _generate_launch_string(
+            argvals,
+            argvals["mode"],
```

Comment:
Should we also refactor `_generate_launch_string()` slightly so that we don't need to pass `argvals` and again `argvals["mode"]`?

Reply:
https://github.com/ansys/pyfluent/issues/2833